src/StudentsController.ts

```diff
@@ -652,4 +652,57 @@ export class StudentsController {
             respondedAt: request.response?.createdAt
         };
     }
+
+    public async sendCertificateNotifications(data: { ids?: string[]; code?: string }): Promise<Result<void>> {
+        const students: Student[] = [];
+
+        if (!data.ids || data.ids.length === 0) {
+            const studentsWithPendingCertificateRequest = await this.getStudentsWithPendingCertificateRequest();
+            if (studentsWithPendingCertificateRequest.length === 0) {
+                return Result.fail(new ApplicationError("error.schoolModule.noPendingCertificateRequest", "There are no students with pending certificate requests."));
+            }
+
+            students.push(...studentsWithPendingCertificateRequest);
+        } else {
+            const idsWithoutDuplicates = Array.from(new Set(data.ids));
+            const recipients = (await Promise.all(idsWithoutDuplicates.map((id) => this.getStudent(id)))).filter((recipient) => recipient !== undefined);
+            if (recipients.length !== data.ids.length) return Result.fail(new ApplicationError("error.schoolModule.studentNotFound", "One or more students were not found."));
```

Comment:
Otherwise, an `"One or more students were not found."` error is thrown if an ID is contained more than once within the array even though there is a matching student.

```suggestion
            const idsWithoutDuplicates = Array.from(new Set(data.ids));
            const recipients = (await Promise.all(idsWithoutDuplicates.map((id) => this.getStudent(id)))).filter((recipient) => recipient !== undefined);
            if (recipients.length !== idsWithoutDuplicates.length) return Result.fail(new ApplicationError("error.schoolModule.studentNotFound", "One or more students were not found."));
```